examples/flutter_firebaseauth/lib/authentication/authentication_service.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class FirebaseAuthService {
```

Comment:
Thoughts on renaming this to `AuthService` or `AuthRepository`? I think it'd be better to have a generic Authentication Service and hide the underlying implementation.

Reply:
Yeah it can be done, somehow it got stuck in my head that a repository is for when you get data from your own database and a service is for external sources. I think the better solution is to have a service in front of a repository but it might be overkill a simple example like this. What do you think?

Reply:
Yeah i think having a UserRepository with signIn and signOut methods would be good for this example. What do you think?